packages/react-sdk/src/components/image-dev.stories.tsx

```diff
@@ -0,0 +1,169 @@
+  // Just emulate like we really resize the image
+  const params = new URLSearchParams();
+  params.set("width", `${width}`);
+  params.set("quality", `${quality}`);
+  return `${src}?${params.toString()}`;
+};
+
+const imageLoader = USE_CLOUDFLARE_IMAGE_TRANSFORM
+  ? cloudflareImageLoader
+  : localImageLoader;
+
+const ImageBase: ComponentStory<
+  React.ForwardRefExoticComponent<
+    ImageProps & { style?: HTMLAttributes<"img">["style"] }
+  >
+> = (args) => {
+  warnOnce(
+    !(
+      imageMeta.defaultStyle?.maxWidth?.value === 100 &&
+      imageMeta.defaultStyle?.maxWidth?.unit === "%"
+    ),
+    "We expect that the image default style has maxWidth: 100%"
+  );
+
+  warnOnce(
+    imageMeta.defaultStyle?.display?.value !== "block",
+    "We expect that the image default style has display: block"
+  );
+
+  const style = {
+    maxWidth: "100%",
+    display: "block",
+    ...args.style,
+  };
+
+  return (
+    <ImagePrimitive
+      {...args}
+      optimize={true}
+      loader={imageLoader}
+      style={style}
+    />
+  );
+};
+
+/**
+ * Load images depending on image width and device per pixel ratio.
+ **/
+export const FixedWidthImage: ComponentStory<React.FunctionComponent> = () => (
+  <ImageBase src={imageSrc} width="300" height="400" />
+);
+
+/**
+ * Preserve ratio using object-fit: cover. Load images depending on image width and device per pixel ratio.
+ **/
+export const FixedWidthImageCover: ComponentStory<
+  React.FunctionComponent
+> = () => (
+  <ImageBase
+    src={imageSrc}
+    width="300"
+    height="400"
+    style={{ objectFit: "cover" }}
+  />
+);
+
+/**
+ * Load images depending on the viewport width.
+ **/
+export const UnknownWidthImage: ComponentStory<
+  React.FunctionComponent
+> = () => <ImageBase src={imageSrc} />;
+
+/**
+ * Fit width of the parent container, has own aspect-ratio and object-fit=cover.
+ * Load images depending on the viewport width.
+ **/
+export const AspectRatioImage: ComponentStory<React.FunctionComponent> = () => (
+  <div style={{ width: "50%" }}>
+    <ImageBase
+      src={imageSrc}
+      style={{ aspectRatio: "2/1", objectFit: "cover", width: "100%" }}
+    />
+  </div>
+);
+
+/**
+ * Fill width and height of the relative parent container, object-fit=cover. Load images depending on the viewport width.
+ **/
+export const FillParentImage: ComponentStory<React.FunctionComponent> = () => (
+  <div style={{ width: "50%", aspectRatio: "2/1", position: "relative" }}>
+    <ImageBase
+      src={imageSrc}
+      style={{
+        objectFit: "cover",
+        position: "absolute",
+        width: "100%",
+        height: "100%",
+      }}
+    />
+  </div>
+);
+
+/**
+ * "sizes" attribute explicitly equal to 100vw allowing to skip the default behavior.
+ * See DEFAULT_SIZES in the Image component. Load images depending on the viewport width.
+ **/
+export const HeroImage: ComponentStory<React.FunctionComponent> = () => (
```

Comment:
Perhaps no need to add types at all?

Reply:
Could be usefull later to add args or params

Reply:
`i.e. HeroImage.params = {docs: {...}}`